src/drivers/TwiMaster.cpp

```diff
@@ -9,21 +9,27 @@ using namespace Pinetime::Drivers;
 // TODO use DMA/IRQ
 
 TwiMaster::TwiMaster(const Modules module, const Parameters& params) : module {module}, params {params} {
+  mutex = xSemaphoreCreateBinary();
```

Comment:
The initialization of the mutex was previously done in the ctor(), but I moved it to Init() in 6d524ebea2c97e309633d5e01c3a1e37c182f27d. The goal was to avoid [Static Initialization Order Fiasco](https://en.cppreference.com/w/cpp/language/siof) (see [this](https://github.com/JF002/InfiniTime/pull/415#issuecomment-859004238). 
I also wanted to ensure FreeRTOS was correctly initialized before instantiating mutexes and other objects. 

Can we ensure this won't cause any issue?

Reply:
There is a separate function called `xSemaphoreCreateBinaryStatic()`, so I'd assume that one isn't static and works correctly.

Reply:
I can't say I completely understand the issue, but this doesn't access global state so it doesn't apply right? I dont think we shoud be afraid of using constructors either because they make the classes safe to use right away, and I would even like to move more stuff to the constructors to reduce the Init functions in code that can be avoided.

Reply:
Yes, of course, most of the time, we shouldn't need an Init() function and use the ctor() to initialise the whole object. However, in this case, the ctor is called by the Startup code, before main() is called.
When you look at the code of `xSemaphoreCreateBinary()` you see it access to variables from FreeRTOS and even calls `pvPortMalloc()`.

I'm not sure completely sure that it's ok that call freertos function before the scheduler is init'd and running.
It might work, as previously, all mutexes were initialized in the ctro(), but I moved them when I was chasing memory corruption issue (which, in the end, were not caused by those initializations in the ctor() so...).

Reply:
I moved it back so we don't need to worry about it now.